File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.java
class Solution {
    
    boolean[] isComposite = new boolean[3000];
    
    void findCompoite(){
        for(int i=2;i<3000;i++){
            for(int j=i*i;j<3000;j+=i){
                if(isComposite[j])
                    continue;
                isComposite[j] = true;
            }
        }
        isComposite[2] = false;
    }
        
        
    public int solution(int[] nums) {
        findCompoite();
        int answer=0;
        
        for(int i=0; i<nums.length-2;i++){
            for(int j=i+1; j<nums.length-1;j++){
                for(int k=j+1; k<nums.length;k++){
                    if(!isComposite[ (nums[i]+nums[j]+nums[k]) ]){
                        answer ++;
                    }
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function isPrime(num){
    if(num<2) return false
    if(num===2) return true
    if(num%2==0) return false
    
    const sqrt = Math.floor(Math.sqrt(num))
    for(let i=3; i<=sqrt; i+=2)
        if(num%i===0) 
            return false
    return true
}

function solution(nums) {
    const numsSize = nums.length
    let answer = 0
    
    for(let a=0; a<numsSize-2;a++){
        for(let b=a+1; b<numsSize-1;b++){
            for(let c=b+1; c<numsSize; c++){
                const aVal = nums[a]
                const bVal = nums[b]
                const cVal = nums[c]
                
                if(isPrime(aVal+bVal+cVal))
                    answer++
            }
        }
    }
    
    return answer
}